swap/src/network/tor_transport.rs

```diff
@@ -87,43 +72,33 @@ fn to_address_string(multi: Multiaddr) -> Result<String> {
             ))
         }
         // Deal with non-onion addresses
-        Some(Protocol::Ip4(addr)) => Some(format!("{}", addr)),
-        Some(Protocol::Ip6(addr)) => Some(format!("{}", addr)),
-        Some(Protocol::Dns(addr)) => Some(format!("{}", addr)),
-        Some(Protocol::Dns4(addr)) => Some(format!("{}", addr)),
-        _ => None,
-    }
-    .ok_or_else(|| {
-        anyhow!(
-            "Could not format address {}. Please consider reporting this issue. ",
-            multi
-        )
-    })?;
-
-    let port_string = match protocols.next() {
-        Some(Protocol::Tcp(port)) => Some(format!("{}", port)),
-        Some(Protocol::Udp(port)) => Some(format!("{}", port)),
-        _ => None,
+        Some(Protocol::Ip4(addr)) => format!("{}", addr),
+        Some(Protocol::Ip6(addr)) => format!("{}", addr),
+        Some(Protocol::Dns(addr)) => format!("{}", addr),
+        Some(Protocol::Dns4(addr)) => format!("{}", addr),
+        _ => return Err(TransportError::MultiaddrNotSupported(multi)),
```

Comment:
I'm curious if this error will show up somewhere in the logs or if libp2p _silently_ falls back to the next transport? 
I'm curious because when this function initially was written I could not think of more or other protocols we want to support here.

Reply:
Yep, the fallback is silent, it happens here: https://github.com/libp2p/rust-libp2p/blob/762f4e35e72dbca4ce82dd5099dad7caf3016650/core/src/transport/choice.rs#L46-L60

Are you ok with the fallback being silent?

Reply:
Sure, it just means that we won't learn about other protocols we might have missed here.

Reply:
We can add more logging here if that is what you are asking. Is your concern that a user might expect things to run via tor but because of us not handling some esoteric thing here it actually ends up being on clearnet?

That sounds more like we should either have a switch or a dedicated binary that enforces the Tor transport and removes the regular tcp transport entirely. That will result in an unrecoverable error if our Tor transport cannot handle some kind of address.

Reply:
> We can add more logging here if that is what you are asking. Is your concern that a user might expect things to run via tor but because of us not handling some esoteric thing here it actually ends up being on clearnet?

Yes, kinda. When I wrote the initial transport my thinking was: 
`If Tor is enabled, everything should go through Tor. But we don't want to application to be unusable if we have missed a protocol here. In that case, an error message should be printed which is hopefully reported to us. We then fall-back to TCP and continue.`

Maybe this wasn't the best behavior neither. 
Anyway: We can say we only support these protocols for Tor, everything else will go via clearnet. I still wouldn't know which one we are missing :)

Reply:
To be honest, I wouldn't count on users reporting warnings like this ...

For now, falling back silently errs on the side of being as compatible as possible. If we see risks there, I would suggest to default to Tor only and only provide a switch the user can actively trigger with which they give consent to fall back to clearnet if Tor doesn't work for some reason.

Reply:
I think the switch would actually be a good idea - that could even be part of the setup wizzard, something along the lines of "Do you want to allow connection over clearnet (or Tor only)?"

ASB providers do care about anonymity; if the application silently decides to go over clearnet, that is not optimal.

Reply:
Yep totally valid point. Keep in mind though that you are only looking at the CLI's transport here! At this stage, the ASB doesn't even know that it is running over Tor.